ext/pdo_oci/tests/pdo_oci_stmt_getcolumnmeta.phpt

```diff
@@ -0,0 +1,281 @@
+--TEST--
+PDO_OCI: PDOStatement->getColumnMeta
+--SKIPIF--
+<?php # vim:ft=php
+if (!extension_loaded('pdo') || !extension_loaded('pdo_oci')) die('skip not loaded');
+require(dirname(__FILE__) . '/../../pdo/tests/pdo_test.inc');
+PDOTest::skip();
+?>
+--FILE--
+<?php
+
+echo "Preparations before the test\n";
+
+require(dirname(__FILE__) . '/../../pdo/tests/pdo_test.inc');
+try {
+	$db = PDOTest::factory();
+	$db->exec(<<<SQL
+BEGIN
+   EXECUTE IMMEDIATE 'DROP TABLE test';
+EXCEPTION
+   WHEN OTHERS THEN
+      IF SQLCODE != -942 THEN
+	 RAISE;
+      END IF;
+END;
+SQL
+);
+	$db->exec("CREATE TABLE test(id INT)");
+
+	$db->beginTransaction();
+
+	$stmt = $db->prepare('SELECT id FROM test ORDER BY id ASC');
+
+	echo "Test 1. calling function with invalid parameters\n";
+
+	// execute() has not been called yet
+	// NOTE: no warning
+	if (false !== ($tmp = $stmt->getColumnMeta(0)))
+		printf("[002] Expecting false got %s\n", var_export($tmp, true));
+
+	$stmt->execute();
+	// Warning: PDOStatement::getColumnMeta() expects exactly 1 parameter, 0 given in
+	if (false !== ($tmp = @$stmt->getColumnMeta()))
+		printf("[003] Expecting false got %s\n", var_export($tmp, true));
+
+	// invalid offset
+	if (false !== ($tmp = @$stmt->getColumnMeta(-1)))
+		printf("[004] Expecting false got %s\n", var_export($tmp, true));
+
+	// Warning: PDOStatement::getColumnMeta() expects parameter 1 to be int, array given in
+	if (false !== ($tmp = @$stmt->getColumnMeta(array())))
+		printf("[005] Expecting false got %s\n", var_export($tmp, true));
+
+	// Warning: PDOStatement::getColumnMeta() expects exactly 1 parameter, 2 given in
+	if (false !== ($tmp = @$stmt->getColumnMeta(1, 1)))
+		printf("[006] Expecting false got %s\n", var_export($tmp, true));
+
+	// invalid offset
+	if (false !== ($tmp = $stmt->getColumnMeta(1)))
+		printf("[007] Expecting false because of invalid offset got %s\n", var_export($tmp, true));
+
+	echo "Test 2. testing return values\n";
```

Comment:
What about testing the type of some functions like 'select count(*) from dual', or use to_char or to_date?

Reply:
good idea